### PR TITLE
Error message for string concat

### PR DESCRIPTION
**Pull request by syukronrm**
Related issue #985 

Produce hint when trying to use `+` and `+.` operator for string:
```
error: Type mismatch
  ┌─ /home/me/projects/gleam/experiments/test_str_concat/src/test_str_concat.gleam:4:9
  │
4 │   100 + "100"
  │         ^^^^^

The + operator expects arguments of this type:

    Int

But this argument has this type:

    String

Hint: String can be concatted using `string.concat` from stdlib


===> Unable to compile Gleam project
```

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -1672,26 +1671,37 @@ fn import_cycle(buffer: &mut Buffer, modules: &[String]) {
...
         _ => None,
     }
 }
+
+fn hint_numeric_message(alt: &str, type_: &str) -> String {
+    format!("the {} operator can be used with {}s\n", alt, type_)
+}
+
+fn hint_string_message() -> String {
+    "String can be concatted using `string.concat` from stdlib".to_string()
```
Comment (lpil) [inline src/error.rs:1706]:
```suggestion
    "Strings can be joined using the `append` or `concat` functions from
the `gleam/string` module".to_string()
```

**Review by lpil — COMMENTED** [review]
Looks great! Would you mind updating the changelog as well? Thank you

**Comment by syukronrm** [comment]
Updated!

**Review by lpil — APPROVED** [review]
Wonderful! Thank you

**Pull request merged**